code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -59,18 +59,18 @@
 	var/timer = 450
 
 /obj/machinery/doomsday_device/process()
-	if(z != ZLEVEL_STATION)
-		minor_announce("DOOMSDAY DEVICE OUT OF RANGE OF STATION, ABORTING", "ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4", 1)
+	var/turf/T = get_turf(src)
+	if(T && T.z != ZLEVEL_STATION)

```

Comment:
Probably does not matter either way, but 

if(!T || T.z != ZLEVEL_STATION)

would catch edgecases like ending up in nullspace